sound/soc/intel/common/soc-acpi-intel-jsl-match.c

```diff
@@ -6,16 +6,33 @@
  *
  */
 
+#include <linux/module.h>
 #include <sound/soc-acpi.h>
-#include <sound/soc-acpi-intel-match.h>
```

Comment:
this doesn't look right. We have the same set of headers included in all soc-acpi....-match files.

So either it's not needed, or it's needed in all cases.

Reply:
I just tested and I see no problems with make ARCH=arm allmodconfig?

Reply:
make ARCH=arm allmodconfig; then you can see in the .config CONFIG_SND_SOC_ACPI_INTEL_MATCH=m, now make a build, it takes quite long on my build machine to finish "make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf -j4".

With x86 build, it does not need #include <sound/soc-acpi-intel-match.h> so I removed it, this header file further includes the module.h, so it does not fail for arm build for all other files under the same dir except this patch.

Reply:
like I said, things work fine for me with ARCH=arm on topic/sof-dev with the upstream code. You must be using an older environment?

Reply:
Sorry for the confusion, the current code on sof-dev is OK, there is no problem.
This patch removed include <sound/soc-acpi-intel-match.h> since v1 which caused the build error for ARM config, so I added the module.h which is the only necessary include.

FYI, https://chromium-review.googlesource.com/c/chromiumos/third_party/kernel/+/1995884/

Reply:
Hi, @plbossart , morning, in rethinking the #include <sound/soc-acpi-intel-match.h>, I feel it's a correct practice to include this header in it's definition file, so added it back in b84cc98.